bin/propolis-server/src/lib/stats/mod.rs

```diff
@@ -217,3 +242,151 @@ pub(crate) async fn track_vcpu_kstats(
         );
     }
 }
+
+/// Track kstats required to publish network interface metrics for this instance.
+#[cfg(any(test, not(target_os = "illumos")))]
+pub(crate) async fn track_network_interface_kstats(
+    log: &Logger,
+    _: &KstatSampler,
+    _: &InstanceProperties,
+    _: NetworkInterfaceIds,
+) {
+    slog::error!(
+        log,
+        "network interface stats are not supported on this platform"
+    );
+}
+
+/// Track kstats required to publish network interface metrics for this instance.
+#[cfg(all(not(test), target_os = "illumos"))]
+pub(crate) async fn track_network_interface_kstats(
+    log: &Logger,
+    sampler: &KstatSampler,
+    properties: &InstanceProperties,
+    interface_ids: NetworkInterfaceIds,
+) {
+    let nics = InstanceNetworkInterfaces::new(properties, interface_ids);
+    let details = oximeter_instruments::kstat::CollectionDetails::never(
+        NETWORK_INTERFACE_SAMPLE_INTERVAL,
+    );
+    let interface_id = nics.target.interface_id;
+    if let Err(e) = sampler.add_target(nics, details).await {
+        slog::error!(
+            log,
+            "failed to add network interface targets, \
+            network interface stats will be unavailable";
+            "network_interface_id" => %interface_id,
```

Comment:
At this point, this UUID is `Uuid::nil()`, right? We probably want to print the list of actual UUIDs, in `nics.interface_ids`, I think.